src/lib.rs

```diff
@@ -525,6 +525,31 @@ pub trait Parser<'a, I: Input<'a>, O, E: ParserExtra<'a, I> = extra::Default>:
         }
     }
 
+    /// Works the same as [`Parser::map`], but the second argument for the mapper F is the parser's
+    /// current context.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use chumsky::{prelude::*, error::Simple};
+    ///
+    /// // let x = empty::<char, _>().to(0usize).map_with_ctx(|ind: usize, ctx: &()| -> usize { ind });
+    /// // let indent = empty::<&str, extra::Default>().to(0usize).map_with_ctx(|ind: usize, ctx: ()| ind);
+    /// let indent = text::whitespace::<char, &str, extra::Default>().count(); //.map_with_ctx(|ind: usize, ctx: usize| ctx + ind);
```

Comment:
I'm not sure I understand this example.

Reply:
I didn't actually finish the example, I wanted to make sure you were interested in including a method like this before I spent time on the docs. I can clean it up before removing WIP though, definitely